pippy/PipelineStage.py

```diff
@@ -476,7 +476,11 @@ def _send_activations(
                 )
                 peer_rank = self.stage_index_to_group_rank[dst]
                 work = dist.isend(
-                    out,
+                    # HACK: we convert DTensor to regular tensor here for it to
+                    # work with send ops. DTensor may show up in PP + TP cases.
+                    out.to_local()
+                    if isinstance(out, torch.distributed._tensor.DTensor)
```

Comment:
I think we should understand why isend would get a DTensor, if Pipeline split each transformerblock it should not get DTensor as inputs